Open Judge System/Data/OJS.Data/OjsDbContext.cs

```diff
@@ -234,5 +252,37 @@ private int SaveChangesWithTracingDbExceptions()
                 throw;
             }
         }
+
+        private async Task<int> SaveChangesWithTracingDbExceptionsAsync()
+        {
+            try
+            {
+                return await base.SaveChangesAsync();
+            }
+            catch (DbUpdateException ex)
+            {
+                Exception currentException = ex;
+                while (currentException != null)
+                {
+                    Trace.TraceError(currentException.Message);
+                    currentException = currentException.InnerException;
+                }
+
```

Comment:
You can extract method `HandleDbUpdateException` and use it both sync and async versions.

Reply:
Done